packages/react/src/useSignIn.ts

```diff
@@ -0,0 +1,83 @@
+import { Address } from '@solana/addresses';
+import {
+    SolanaSignIn,
+    SolanaSignInFeature,
+    SolanaSignInInput,
+    SolanaSignInOutput,
+} from '@solana/wallet-standard-features';
+import {
+    getWalletAccountFeature,
+    getWalletFeature,
+    UiWallet,
+    UiWalletAccount,
+    UiWalletHandle,
+} from '@wallet-standard/ui';
+import {
+    getOrCreateUiWalletAccountForStandardWalletAccount_DO_NOT_USE_OR_YOU_WILL_BE_FIRED,
+    getWalletAccountForUiWalletAccount_DO_NOT_USE_OR_YOU_WILL_BE_FIRED,
+    getWalletForHandle_DO_NOT_USE_OR_YOU_WILL_BE_FIRED,
+} from '@wallet-standard/ui-registry';
+import { useCallback } from 'react';
+
+type Input = SolanaSignInInput;
+type Output = Omit<SolanaSignInOutput, 'account' | 'signatureType'> &
+    Readonly<{
+        account: UiWalletAccount;
+    }>;
+
+/**
+ * Returns a function you can call to sign in to a domain
+ */
+export function useSignIn(uiWalletAccount: UiWalletAccount): (input?: Omit<Input, 'address'>) => Promise<Output>;
+export function useSignIn(uiWallet: UiWallet): (input?: Input) => Promise<Output>;
+export function useSignIn(uiWalletHandle: UiWalletHandle): (input?: Input) => Promise<Output> {
+    const signIns = useSignIns(uiWalletHandle);
+    return useCallback(
+        async input => {
+            const [result] = await signIns(input);
+            return result;
+        },
+        [signIns],
+    );
+}
+
+function useSignIns(
```

Comment:
I think we can do away this plural API, seems very unlikely to be used and certainly doesn't work today.

Reply:
Oh I see, this isn't exported anyway, this is just internally how it's implemented because the Wallet Standard is this way. Carry on!